frontend/src/pages/signIn/SignInController.jsx

```diff
@@ -0,0 +1,54 @@
+import { useContext, useState } from 'react'
+import { useNavigate } from 'react-router'
+import { AuthContext } from '../../contexts/AuthProvider'
+import { postData } from '../../functions'
+import validationEmail from '../../functions/validateEmail'
+import SignInView from './SignInView'
+
+const SignInController = () => {
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+    const [error, setError] = useState('')
+    const [loading, setLoading] = useState(false)
+
+    const navigate = useNavigate()
+
+    const authContext = useContext(AuthContext)
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+        setLoading(true)
+        if (email === '' || password === '') {
+            setError('Please fill out all fields')
+        } else if (!validationEmail(email)) {
+            setError('Invalid email address')

```

Comment:
loading needs to be set as false if there is a form validation error, otherwise the sign in button will be permanently loading